### PR TITLE
add new feature: addBabelPluginOutSideOfApp and update snapshot

### PR DESCRIPTION
**Pull request by Terryand**
- update getBabelLoader as cra2/3 acturally has two babel-loader relues. One is for appSrc and the other is for outside of app( like node_modules ).Different them by include/exclude options;
- add new feature: addBabelPluginOutSideOfApp. Now you can add plugin to the babel-loader which is for outside of app. Of course, test suits is updated together
- update snapshots: now the snapshots is more like to real cra2/3 webpack-config;

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,9 @@
 node_modules/
 coverage/
+
+# IDE
+.idea
+.vscode
+
+# Disk
+.DS_Store
```
[the remark this answers is not in the document]
Comment (Terryand) [reply .gitignore]: Thanks.
I have renamed to addExternalBabelPlugin/addExternalBabelPlugins. Also I added examples in `api.md`.
My English is not so good. So the docs might be not good enough. Could you help me modify the `api.md` or `readme.md` if you find some thing wrong ?

File: api.md [hunk 2]
```diff
@@ -347,3 +347,29 @@ const { addTslintLoader } = require("customize-cra");
 
 module.exports = override(addTslintLoader());
 ```
+
+### addExternalBabelPlugin(plugin)
+
+Just like `addBabelPlugin`, you can add a babel plugin for 'external' babel-loader by this way. Actually in create-react-app v2/v3, it has two rules for babel-loader. One is for the codes which in `appSrc` and the other is for the codes which 'outside of your app' like `node_modules`. If you need customize the second babel-loader plugins, you can use this method. Reminder again, Whatever you pass for `plugin` will be added to Babel's `plugins` array. Consult their docs for more info.
```
Comment (with-heart) [inline api.md]:
```suggestion
`create-react-app` actually has two rules in its `webpack` config for `babel-loader`: one for code in `addSrc` (`src/` by default) and the other for code `external` to that folder (like `node_modules`). You can add plugins to the `external` loader using `addExternalBabelPlugin` in the same way you'd use `addBabelPlugin`.
```

File: api.md [hunk 3]
```diff
@@ -347,3 +347,29 @@ const { addTslintLoader } = require("customize-cra");
...
+
+### addExternalBabelPlugins(plugins)
+
+A simple helper that calls `addExternalBabelPlugin` for each plugin you pass in here. Make sure you use the spread operator when using this, for example
```
Comment (with-heart) [inline api.md]:
```suggestion
A simple helper that calls `addExternalBabelPlugin` for each plugin passed. 

Note: Make sure to use the spread operator if adding multiple plugins.
```

**Comment by Terryand** [comment]
That's ok and thanks a lot for these suggestions.
The `api.md` is be modified. Look forward to merging.

**Comment by with-heart** [comment]
Great work @Terryand!

**Comment by with-heart** [comment]
Published as [`v0.3.0`](https://github.com/arackaf/customize-cra/releases/tag/v0.3.0) :)

**Comment by with-heart** [comment]
Hey @Terryand, thanks again! I've added you to the project's contributors list :)

**Pull request merged**